server/app/services/agent/message_handler.rb

```diff
@@ -188,13 +194,17 @@ def on_node_stat(grid, data)
       node = grid.host_nodes.find_by(node_id: data['id'])
       return unless node
       data = fixnums_to_float(data)
+      time = data['time'] ? Time.parse(data['time']) : Time.now.utc
```

Comment:
I agree, it would be more reliable if agent sends the time. Could be also a separate PR.